cltk/lemmatize/middle_high_german/backoff.py

```diff
@@ -0,0 +1,77 @@
+"""Module for lemmatizing Middle High German
+"""
+
+__author__ = ['Clément Besnier <clemsciences@aol.com>', ]
+__license__ = 'MIT License. See LICENSE.'
+
+import os
+from typing import List
+
+from cltk.lemmatize.backoff import IdentityLemmatizer, DictLemmatizer, UnigramLemmatizer
+
+from cltk.utils.file_operations import open_pickle
+
+
+def _randomize_data(train: List[list], seed: int):
+    import random
+    random.seed(seed)
+    random.shuffle(train)
+    pos_train_sents = train[:4000]
+    lem_train_sents = [[(item[0], item[1]) for item in sent] for sent in train]
+    train_sents = lem_train_sents[:4000]
+    test_sents = lem_train_sents[4000:5000]
+
+    return pos_train_sents, train_sents, test_sents
+
+
+class BackoffMHGLemmatizer:
+    """Suggested backoff chain; includes at least on of each
+    type of major sequential backoff class from backoff.py
+    """
+
+    models_path = os.path.normpath(get_cltk_data_dir() +
+                                   '/middle_high_german/model/middle_high_german_models_cltk/lemmata/backoff')
+
+    def __init__(self, train: List[list] = None, seed: int = 3, verbose: bool = False):
+        self.models_path = BackoffMHGLemmatizer.models_path
+
+        missing_models_message = "BackoffMHGLemmatizer requires the ```middle_high_german_models_cltk``` " \
+                                 "to be in cltk_data. Please load this corpus."
+        self.train = train
+        self.seed = seed
+        self.verbose = verbose
+
+        self.token_to_lemmata = []
+        self.lemma_to_tokens = []
+        self.sentences = []
+
+        try:
+            self.lemma_to_tokens = open_pickle(os.path.join(self.models_path, "lemma_to_tokens.pickle"))
+            self.token_to_lemmata = open_pickle(os.path.join(self.models_path, "token_to_lemma.pickle"))
+            # self.sentences = open_pickle(os.path.join(self.models_path, "sentences.pickle"))
```

Comment:
If you are not going to include the TrainLemmatizer in this default backoff setup, you should remove all of the sentence-related code; so anything having to do with setting up train/test data, etc. You also have to take out the evaluate method (for now) because it builds its test set from sentence data. (Something that could be revisited; e.g. an alternate way of adding a test set for langs that do not have sentence data.)

Reply:
I removed the sentence-related code which is not used. I will probably add an evaluation of the lemmatizer in an other pull request.

Reply:
What remains to be done?

Reply:
resolving because outdated and no longer applies